version: '3.9'

services:
  db:
    image: mysql:8.0
    restart: always
    environment:
      MYSQL_DATABASE: XUSSIO_DB
      MYSQL_USER: xussio_admin
      MYSQL_PASSWORD: 4dm!n
      MYSQL_ROOT_PASSWORD: $$i0
    ports:
      - "3306:3306"
    volumes:
      - mysql_data:/var/lib/mysql
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "xussio_admin", "-p4dm!n"]
      interval: 10s
      timeout: 5s
      retries: 5
    
  ad_db:
    image: mysql:8.0
    container_name: ad_db
    environment:
      MYSQL_ROOT_PASSWORD: ad_$$i0
      MYSQL_DATABASE: XUSSIO_AD 
      MYSQL_USER: xussioad_admin
      MYSQL_PASSWORD: AD4dm!n
    ports:
      - "3307:3306"
    volumes:
      - ad_data:/var/lib/mysql
<<<<<<< HEAD
      - ./init_ad.sql:/docker-entrypoint-initdb.d/init_ad.sql
=======
      - ./init_ad.sql:/docker-entrypoint-initdb.d/init.sql
>>>>>>> f1d7af53b2ec1c899a873523aef130c85bdfcbd5

  django-web:
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    working_dir: /app
    volumes:
      - .:/app
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
    environment:
      DATABASE_HOST: db
      DATABASE_NAME: XUSSIO_DB
      DATABASE_USER: xussio_admin
      DATABASE_PASSWORD: 4dm!n

volumes:
  mysql_data:
  ad_data:
